llm/llama-3_1/README.md

```diff
@@ -0,0 +1,300 @@
+
+### Step 0: Bring your infra
+
+Install SkyPilot on your local machine:
+
+```bash
+pip install 'skypilot-nightly[all]'
+```
+
+Pick one of the following depending on what infra you want to run Llama 3.1 on:
+
+**If your local machine is a GPU node**: use this command to up a lightweight kubernetes cluster:
+
+```bash
+sky local up
+```
+
+**If you have a Kubernetes GPU cluster** (e.g., on-prem, EKS / GKE / AKS / ...):
+
+```bash
+# Should show Enabled if you have ~/.kube/config set up.
+sky check kubernetes
+```
+
+**If you want to use clouds** (e.g., reserved instances): 12+ clouds are supported:
+
+```bash
+sky check
+```
+
+See [docs](https://skypilot.readthedocs.io/en/latest/getting-started/installation.html) for details.
+
+### Step 1: Get a GPU dev node (pod or VM)
+
+> **Tip:** Skip directly to Step 3 to get the SkyPilot YAML to deploy Llama 3.1 on your infra.
+
+One command to get a GPU dev pod/VM:
+```bash
+sky launch -c llama --gpus A100-80GB:8
+```
+If you are using local machine or Kubernetes, the above will create a pod. If you are using clouds, the above will create a VM.
+
+You can add a `-r / --retry-until-up` flag to have SkyPilot auto-retry to guard against out-of-capacity errors.
+
+
+> **Tip:** Vary the `--gpus` flag to get different GPU types and counts. For example, `--gpus H100:8` gets you a pod with 8x H100 GPUs.
+>
+> You can run `sky show-gpus` to see all available GPU types on your infra.
+
+
+Once provisioned, you can easily connect to it to start dev work. Two recommended methods:
+- Open up VSCode, click bottom left, `Connect to Host`, type `llama`
+- Or, SSH into it with `ssh llama`
+
+### Step 2: Inside the dev node, test serving
+
+Once logged in, run the following to install vLLM and run it (which automatically pulls the model weights from HuggingFace):
+```bash
+pip install vllm==0.5.3.post1 huggingface
+
+# Paste your HuggingFace token to get access to Meta Llama repos:
+# https://huggingface.co/collections/meta-llama/llama-31-669fc079a0c406a149a5738f
+huggingface-cli login
+```
+
+We are now ready to start serving.  If you have N=8 GPUs
+```bash
+vllm serve meta-llama/Meta-Llama-3.1-8B-Instruct --tensor-parallel-size 8
+```
+Change the `--tensor-parallel-size` to the number of GPUs you have.
+
+Tip: available model names can be found [here](https://huggingface.co/collections/meta-llama/llama-31-669fc079a0c406a149a5738f) and below.
+- Pretrained:
+  - Meta-Llama-3.1-8B
+  - Meta-Llama-3.1-70B
+  - Meta-Llama-3.1-405B-FP8
+- Instruction tuned:
+  - Meta-Llama-3.1-8B-Instruct
+  - Meta-Llama-3.1-70B-Instruct
+  - Meta-Llama-3.1-405B-Instruct-FP8
+
+
+The full precision 405B model Meta-Llama-3.1-405B requires multi-node inference and is work in progress - join the [SkyPilot community Slack](https://slack.skypilot.co/) for discussions.
+
+If you are wondering what GPU you need for each model, here is the compatibility matrix (applies to both pretrained and instruction tuned models):
```

Comment:
This table is pretty useful. How about putting it into its own section before Step 0? e.g., "What hardware to use for each Llama 3.1 model"?